diff --git a/PythonAPI/pycocotools/coco.py b/PythonAPI/pycocotools/coco.py
index e4182c7..23436f1 100644
--- a/PythonAPI/pycocotools/coco.py
+++ b/PythonAPI/pycocotools/coco.py
@@ -230,7 +230,7 @@ class COCO:
         elif type(ids) == int:
             return [self.imgs[ids]]
 
-    def showAnns(self, anns, draw_bbox=False):
+    def showAnns(self, anns, draw_bbox=False, ax_select=None):
         """
         Display the specified annotations.
         :param anns (array of object): annotations to display
@@ -238,14 +238,17 @@ class COCO:
         """
         if len(anns) == 0:
             return 0
-        if 'segmentation' in anns[0] or 'keypoints' in anns[0]:
+        if 'segmentation' in anns[0] or 'keypoints' in anns[0] or 'bbox' in anns[0]:
             datasetType = 'instances'
         elif 'caption' in anns[0]:
             datasetType = 'captions'
         else:
             raise Exception('datasetType not supported')
         if datasetType == 'instances':
-            ax = plt.gca()
+            if ax_select is not None:
+                ax = ax_select
+            else:
+                ax = plt.gca()
             ax.set_autoscale_on(False)
             polygons = []
             color = []
@@ -256,7 +259,7 @@ class COCO:
                         # polygon
                         for seg in ann['segmentation']:
                             poly = np.array(seg).reshape((int(len(seg)/2), 2))
-                            polygons.append(Polygon(poly))
+                            #polygons.append(Polygon(poly))
                             color.append(c)
                     else:
                         # mask
@@ -294,8 +297,8 @@ class COCO:
                     polygons.append(Polygon(np_poly))
                     color.append(c)
 
-            p = PatchCollection(polygons, facecolor=color, linewidths=0, alpha=0.4)
-            ax.add_collection(p)
+            #p = PatchCollection(polygons, facecolor=color, linewidths=0, alpha=0.4)
+            #ax.add_collection(p)
             p = PatchCollection(polygons, facecolor='none', edgecolors=color, linewidths=2)
             ax.add_collection(p)
         elif datasetType == 'captions':
@@ -438,4 +441,4 @@ class COCO:
         """
         rle = self.annToRLE(ann)
         m = maskUtils.decode(rle)
-        return m
\ No newline at end of file
+        return m
